contracts/oracles/lido/LidoOracle.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.6;
+
+import '@yield-protocol/utils-v2/contracts/access/AccessControl.sol';
+import '@yield-protocol/utils-v2/contracts/cast/CastBytes32Bytes6.sol';
+import '@yield-protocol/utils-v2/contracts/token/IERC20Metadata.sol';
+import '@yield-protocol/vault-interfaces/IOracle.sol';
+import '../../constants/Constants.sol';
+import './IWstETH.sol';
+
+/**
+ * @title LidoOracle
+ * @notice
+ */
+contract LidoOracle is IOracle, AccessControl, Constants {
+    using CastBytes32Bytes6 for bytes32;
+    IWstETH wstETH;
+    bytes32 public immutable wstEthId;
+    bytes32 public immutable stEthId;
+
+    constructor(bytes32 wstEthId_, bytes32 stEthId_) public {
+        wstEthId = wstEthId_;
+        stEthId = stEthId_;
+    }
+
+    function setSource(IWstETH wstETH_) external auth {
+        wstETH = wstETH_;
+    }
+
+    /**
+     * @notice Retrieve the value of the amount at the latest oracle price.
+     */
+    function peek(
+        bytes32 base,
+        bytes32 quote,
+        uint256 baseAmount
+    ) external view virtual override returns (uint256 quoteAmount, uint256 updateTime) {
+        return _peek(base.b6(), quote.b6(), baseAmount);
+    }
+
+    /**
+     * @notice Retrieve the value of the amount at the latest oracle price. Same as `peek` for this oracle.
+     */
+    function get(
+        bytes32 base,
+        bytes32 quote,
+        uint256 baseAmount
+    ) external virtual override returns (uint256 quoteAmount, uint256 updateTime) {
+        return _peek(base.b6(), quote.b6(), baseAmount);
+    }
+
+    function _peek(
+        bytes6 base,
+        bytes6 quote,
+        uint256 baseAmount
+    ) private view returns (uint256 quoteAmount, uint256 updateTime) {
```

Comment:
I doubt this will ever have any child contracts, but should we follow the practice of using internal by default?

`V10 - Only use private to intentionally prevent child contracts from accessing the variable, prefer internal for flexibility.`

Reply:
`internal` would do just fine, although this is a problem common to all of our oracles, I'm noticing now.

I would leave it as `private` for consistency, and then swap it in all oracles for `internal` in a separate PR